src/nupic/research/temporal_memory.py

```diff
@@ -120,394 +224,218 @@ def compute(self, activeColumns, learn=True):
+    @param excitedColumn   (dict)   Generated by excitedColumnsGenerator
+    @param learn           (bool)   Determines if permanences are adjusted
+    @permanenceDecrement   (float)  Amount by which permanences of synapses are decremented during learning.
+    @permanenceIncrement   (float)  Amount by which permanences of synapses are incremented during learning.
+    @param prevActiveCells (list)   Active cells in `t-1`
+
+    @return cellsToAdd (list) A list of predicted cells that will be added to
+                              active cells and winner cells.
+    """
+
+    cellsToAdd = []
+    cell = None
+    for active in excitedColumn["activeSegments"]:
+      newCell = not cell == connections.cellForSegment(active)
+      if newCell:
+        cell = connections.cellForSegment(active)
+        cellsToAdd.append(cell)
+
+      if learn:
+        TemporalMemory.adaptSegment(connections, prevActiveCells,
+                                    permanenceIncrement, permanenceDecrement,
+                                    active)
+
+    return cellsToAdd
 
-      - if predictedSegmentDecrement > 0
-        - for each distal dendrite segment with unconnected
-          activity >=  minThreshold
-          - mark the segment as matching
-          - mark the cell as matching
 
-    Forward propagates activity from active cells to the synapses that touch
-    them, to determine which synapses are active.
-
-    @param activeCells (set)         Indices of active cells in `t`
-    @param connections (Connections) Connectivity of layer
+  @staticmethod
+  def burstColumn(cellsPerColumn, columnDimensions, connections, excitedColumn,
+                  learn, initialPermanence, matchingSegments, maxNewSynapseCount,
+                  permanenceDecrement, permanenceIncrement, prevActiveCells,
+                  prevWinnerCells, random):
+    """
+    @param cellsPerColumn      (int)    Number of cells per column
+    @param columnDimensions    (list)   Dimensions of the column space
+    @param connections         (Object) Connections instance for the tm
+    @param excitedColumn       (dict)   Excited Column instance from excitedColumnsGenerator
+    @param learn               (bool)   Whether or not learning is enabled
+    @param initialPermanence   (float)  Initial permanence of a new synapse.
+    @param matchingSegments    (list)   List of segments matching from lateral input
+    @param maxNewSynapseCount  (int)    The maximum number of synapses added to a segment during learning.
+    @param permanenceDecrement (float)  Amount by which permanences of synapses are decremented during learning.
+    @param permanenceIncrement (float)  Amount by which permanences of synapses are incremented during learning.
+    @param prevActiveCells     (list)   Active cells in `t-1`
+    @param prevWinnerCells     (list)   Winner cells in `t-1`
+    @param random              (object) Tm object used to generate random numbers
```

Comment:
It might be best to just say "Random number generator" rather than specifying that it's associated with a "Tm". (Ditto for the other occurrences of "param random")